admin_guide/configuring_authentication.adoc

```diff
@@ -0,0 +1,526 @@
+validate user names and passwords against a flat file generated using
+http://httpd.apache.org/docs/2.4/programs/htpasswd.html[`*htpasswd*`].
+
+Only MD5 and SHA encryption types are supported. MD5 encryption is recommended,
+and is the default for `htpasswd`. Plaintext, crypt, and bcrypt hashes are not
+currently supported.
+
+The flat file is re-read if its modification time changes, without requiring a
+server restart.
+
+To create the file, run:
+
+****
+`$ htpasswd -c _</path/to/users.htpasswd>_ _<user_name>_`
+****
+
+To add or update a login to the file, run:
+
+****
+`$ htpasswd _</path/to/users.htpasswd>_ _<user_name>_`
+****
+
+To remove a login from the file, run:
+
+****
+`$ htpasswd _</path/to/users.htpasswd>_ -D _<user_name>_`
+****
+
+.Master Configuration Using `HTPasswdPasswordIdentityProvider`
+====
+
+----
+oauthConfig:
+  ...
+  identityProviders:
+  - name: my_htpasswd_provider <1>
+    challenge: true <2>
+    login: true <3>
+    provider:
+      apiVersion: v1
+      kind: HTPasswdPasswordIdentityProvider
+      file: /path/to/users.htpasswd <4>
+----
+<1> This provider name is prefixed to user names to form an identity name.
+<2> When `true`, unauthenticated token requests from non-web clients (like the CLI) are sent a `WWW-Authenticate` challenge header for this provider.
+<3> When `true`, unauthenticated token requests from web clients (like the Management Console) are redirected to a login page backed by this provider.
+<4> File generated using http://httpd.apache.org/docs/2.4/programs/htpasswd.html[`*htpasswd*`].
+====
+
+=== Basic Authentication (Remote) [[BasicAuthPasswordIdentityProvider]]
+
+Set `BasicAuthPasswordIdentityProvider` in the `identityProviders` stanza to
+validate user names and passwords against a remote server using a
+server-to-server Basic authentication request. User names and passwords are
+validated against a remote URL that is protected by Basic authentication and
+returns JSON.
+
+A `401` response indicates failed authentication.
+
+A non-`200` status, or the presence of a non-empty "error" key, indicates an error:
+
+----
+{"error":"Error message"}
+----
+
+A `200` status with a `sub` (subject) key indicates success:
+
+----
+{"sub":"userid"} <1>
+----
+<1> The subject must be unique to the authenticated user and must not be able to be modified.
+
+A successful response may optionally provide additional data, such as:
+
+* A display name using the `name` key. For example:
++
+----
+{"sub":"userid", "name": "User Name", ...}
+----
++
+* An email address using the `email` key. For example:
++
+----
+{"sub":"userid", "email":"user@example.com", ...}
+----
++
+* A preferred user names using the `preferred_username` key. This is useful when the unique, unchangeable subject is a database key or UID, and a more human-readable name exists. This is used as a hint when provisioning the OpenShift user for the authenticated identity. For example:
```

Comment:
@adellape Should this be "A series of preferred user names..." or maybe "A preferred user name..."?